mempool/mempool.go

```diff
@@ -1354,12 +1353,13 @@ func (mp *TxPool) processOrphans(acceptedTx *dcrutil.Tx) []*dcrutil.Tx {
 	var acceptedTxns []*dcrutil.Tx
 
 	// Start with processing at least the passed transaction.
-	processList := list.New()
-	processList.PushBack(acceptedTx)
-	for processList.Len() > 0 {
+	var processList []*dcrutil.Tx
+	processList = append(processList, acceptedTx)
```

Comment:
You can avoid an extra allocation here by doing:
```Go
processList := []*dcrutil.Tx{acceptedTx}
```

Proof:
```
func BenchmarkPR1681(b *testing.B) {
	for i := 0; i < b.N; i++ {
		var list []int
		list = append(list, 1)
		for len(list) > 0 {
			item := list[0]
			list = list[1:]
			_ = item
		}
	}
}

func BenchmarkPR1681LessAlloc(b *testing.B) {
	for i := 0; i < b.N; i++ {
		list := []int{1}
		for len(list) > 0 {
			item := list[0]
			list = list[1:]
			_ = item
		}
	}
}
```

```
BenchmarkPR1681            50000000     8 B/op   1 allocs/op
BenchmarkPR1681LessAlloc   2000000000   0 B/op   0 allocs/op
```

Reply:
updated